src/pgduckdb_options.cpp

```diff
@@ -199,6 +237,69 @@ DECLARE_PG_FUNCTION(cache) {
 	PG_RETURN_BOOL(result);
 }
 
+DECLARE_PG_FUNCTION(cache_info) {
+	ReturnSetInfo *rsinfo = (ReturnSetInfo *)fcinfo->resultinfo;
+	Tuplestorestate *tuple_store;
+	TupleDesc cache_info_tuple_desc;
+	AttInMetadata *attinmeta;
+
+	auto result = pgduckdb::DuckdbGetCachedFile();
+
+	cache_info_tuple_desc = CreateTemplateTupleDesc(3);
+	TupleDescInitEntry(cache_info_tuple_desc, (AttrNumber)1, "cache_file_name", TEXTOID, -1, 0);
+	TupleDescInitEntry(cache_info_tuple_desc, (AttrNumber)2, "remote_path", TEXTOID, -1, 0);
+	TupleDescInitEntry(cache_info_tuple_desc, (AttrNumber)3, "cache_file_size", TEXTOID, -1, 0);
+
+	// We need to switch to long running memory context
+	MemoryContext oldcontext = MemoryContextSwitchTo(rsinfo->econtext->ecxt_per_query_memory);
+	attinmeta = TupleDescGetAttInMetadata(cache_info_tuple_desc);
+	tuple_store = tuplestore_begin_heap(rsinfo->allowedModes & SFRM_Materialize_Random, false, work_mem);
+	MemoryContextSwitchTo(oldcontext);
+
+	char **values = (char **)palloc0((3) * sizeof(char *));
+
+	for (int i = 0; i < result.size(); i++) {
+
+		auto &cache_info = result[i];
+
+		// Local cache file key
+		char *cache_key = (char *)palloc0(cache_info.cache_key.length() + 1);
+		memcpy(cache_key, cache_info.cache_key.c_str(), cache_info.cache_key.length());
+		values[0] = cache_key;
+
+		// Remote path
+		char *remote_path = (char *)palloc0(cache_info.remote_path.length() + 1);
+		memcpy(remote_path, cache_info.remote_path.c_str(), cache_info.remote_path.length());
+		values[1] = remote_path;
+
+		// Cached file size
+		char *file_size = (char *)palloc0(cache_info.file_size.length() + 1);
+		memcpy(file_size, cache_info.file_size.c_str(), cache_info.file_size.length());
+		values[2] = file_size;
+
+		HeapTuple tuple;
+		tuple = BuildTupleFromCStrings(attinmeta, values);
+		tuplestore_puttuple(tuple_store, tuple);
+		heap_freetuple(tuple);
+
+		pfree(file_size);
+		pfree(remote_path);
+		pfree(cache_key);
```

Comment:
It's too bad that PG doesn't allow to create tuple with no copy/realloc :(

Reply:
Changes so we don't need to pmalloc/pfree for intermediate results.